src/stream/src/executor/source/fs_source_executor.rs

```diff
@@ -439,17 +439,18 @@ impl<S: StateStore> FsSourceExecutor<S> {
                         let state: Vec<(SplitId, SplitImpl)> = mapping
                             .iter()
                             .flat_map(|(id, offset)| {
-                                let origin_split =
-                                    self.stream_source_core.stream_source_splits.get_mut(id);
-
-                                origin_split.map(|split| {
-                                    split.update_in_place(offset.clone())?;
-                                    Ok::<_, anyhow::Error>((id.clone(), split.clone()))
-                                })
+                                self.stream_source_core.latest_split_info.get_mut(id).map(
+                                    |origin_split| {
+                                        origin_split.update_in_place(offset.clone())?;
+                                        Ok::<_, anyhow::Error>((id.clone(), origin_split.clone()))
+                                    },
+                                )
                             })
                             .try_collect()?;
 
-                        self.stream_source_core.state_cache.extend(state);
+                        self.stream_source_core
+                            .updated_splits_in_epoch
+                            .extend(state);
```

Comment:
We can see they are updated in the same way on data chunk. Plan to unify them later.